### ISSUE TITLE
Server reported an error when the client connected using mqtt v5.0

### ISSUE THREAD

**Issue by wind-c** [mochi-co/mqtt#178]
WRN error="read connection: client identifier not valid" listener=tcp
This error probably introduced by fix #171 #172.

In addition, if Username is not specified during client connection, the server reports an error：
WRN error="protocol violation: username flag set but no value" listener=tcp

I used mqttx of EMQX test and found this two problems.

**Comment by ffa500** [comment]
I don't think this is related to my fix #171 #172. 

Using the latest version of [MQTT X 1.9.1](https://www.emqx.com/en/products/mqttx) in combination with the Client Ids autogenerated by the tool or my own ones, I cannot reproduce the first error "read connection: client identifier not valid".

I can reproduce the second error in case the Username field in MQTT X is left empty. The MQTT X Tool **always** sets the User Name Flag to 1 in the CONNECT packet, even if the Username field is left empty. When receiving such a CONNECT packet, Mochi reports a protocol violation according to [MQTT-3.1.2-17]: _If the User Name Flag is set to 1, a User Name MUST be present in the Payload [MQTT-3.1.2-17]._

I'm not sure if this interpretation is quite correct, as the User Name _is_ present in the CONNECT packet, although it only consists of the two bytes [0 0], indicating an empty UTF-8 encoded string. One might interpret [MQTT-3.1.2-17] to also allow an empty User Name if the User Name Flag is set to 1.

Other publicly available brokers such as HIVEMQ (broker.hivemq.com) and EMQX (broker.emqx.io) accept such CONNECT packets.

For reproduction, please find next the raw bytes of the CONNECT packet sent by the MQTT X tool and received by Mochi:

```
[
0 4 77 81 84 84 5   // MQTT 5

130   // byte 8 connect flags User Name Flag = 1, Clean Start = 1

0 60  // keep alive 60

5   // Properties  VBI length 5

17 0 0 0 0  // session expiry

// Connect Payload

0 14 109 113 116 116 120 95 51 48 100 50 48 102 97 56  // Client Identifier mqttx_30d20fa8

0 0  // User Name 2-byte UTF-8 length = 0, empty username
]
```

**Comment by mochi-co** [comment]
Thank you very much for your investigation @ffa500 and for raising it @wind-c! I have implemented a change in v2.2.4 which treats Connect packets with the username flag but no username as valid as long as the LSB+MSB bytes are correct [0,0] (as in the example packet).

Please let me know if it resolves the issue!

**Comment by ffa500** [comment]
Works like a charm!
